tsdb/querier.go

```diff
@@ -347,6 +353,9 @@ func postingsForMatcher(ix IndexReader, m *labels.Matcher) (index.Postings, erro
 
 	var res []string
 	for _, val := range vals {
+		if ctx.Err() != nil {
```

Comment:
This seems like overkill to me - how long can we spend checking individual values?